elixir-phoenix-deployement.md

```diff
@@ -0,0 +1,162 @@
+    - Add erlang plugin to asdf: `asdf plugin-add erlang`
+    - Install Erlang: `asdf install erlang latest`
+    - Install Elixir: `asdf install elixir latest`
+    - Define which Elixir version to use `asdf global elxir <version>`
+
+- Instsall Nodejs using nvm
+    - `wget -qO- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.3/install.sh | bash` (from https://github.com/nvm-sh/nvm#installing-and-updating)
+    - run `nvm install node`
+
+- Clone and run the Phoenix application
+    - Clone the application, e.g. `git clone https://github.com/dwyl/hits.git`
+    - Make sure to have all the environemt variables for the application defined
+        - e.g. for the secret key: `mix phx.gen.secret` then `export SECRET_KEY_BASE=<secret>`
+    - Compile assets (see https://hexdocs.pm/phoenix/deployment.html#compiling-your-application-assets) 
+        - `npm run deploy --prefix ./assets`
+        - `mix phx.digest`
+    - Start the server with `Mix`: `MIX_ENV=prod mix phx.server`
+
+- Another way to run the server is to use `mix release`: https://hexdocs.pm/phoenix/releases.html
+
+#### FreeBSD/OpenBSD
+
+I've also been investigating how to run
+an Elixir/Phoenix applicaiton on FreeBSD (and OpenBSD)
+
+Linode provides a way to create server from image,
+however the backup system won't support server running FreeBSD:
+![image](https://user-images.githubusercontent.com/6057298/83650091-d931c980-a5af-11ea-8ed1-ffc693d79e41.png)
+
+The following guide explain how to install FreeBSD on Linode:
+https://www.linode.com/docs/tools-reference/custom-kernels-distros/install-freebsd-on-linode/
+
+I've also tested the installation on one of my machine: https://github.com/SimonLab/FreeBSD-installation
+
+The idea is then to use the FreeBSD package manager to install Elixir and Erlang:
+- `pkg install erlang`
+- `pkg install elixir`
+
+
+### DigitalOcean
+
+DigitalOcean provides a FreeBSD droplet:
+![image](https://user-images.githubusercontent.com/6057298/83651322-472ac080-a5b1-11ea-8ce3-764cb9fe7927.png)
+
+see https://www.digitalocean.com/products/linux-distribution/freebsd/
+
+However OpenBSD can't be installed directly.
+There are some way we could install it and investigate
+if it can be used safely and without any blockers with DigitalOcean:
+https://dev.to/nabbisen/custom-openbsd-droplet-on-digitalocean-4a9o
+
+see also: https://www.digitalocean.com/community/tutorials/how-to-get-started-with-freebsd
+
+
+## Current Conclusion
+
+After reading and testing some Elixir/Phoenix/Linux/BSD installations
+I can see that Linode (or similar) can be on a longer term a better tool
+to manage the applications.
+
+The simple deployement used above with Ubuntu works well,
+however I still have some research and testing to do especially
+linked to continuous deployment without downtime.
+From reading the chapter 11 "Deploy Your Application to Production" of
```

Comment:
Chapter 11 of the book focusses on using `mix release` which is why I asked if you had read it in: https://github.com/dwyl/learn-devops/issues/59#issuecomment-636433197
> "_Releases are produced with one of two tools—Mix or Distillery._"
![image](https://user-images.githubusercontent.com/194400/83729760-029a3600-a640-11ea-9e00-25bb650c47ef.png)